webapp/src/ts/services/contact-types.service.ts

```diff
@@ -98,6 +98,16 @@ export class ContactTypesService {
     return contactTypesUtils.isPersonType(type);
   }
 
+  async isPerson(contact) {
```

Comment:
It's confusing to have `isPerson` and `isPersonType` exposed simultaneously in this service. It makes someone think we can use `isPersonType` wrongly because a type can be an object.  

Instead, it seems safer to have this function only since it already handles possible datatypes. And then remove `isPersonType(type?) {`

Reply:
Afaic, if you have a contact document and you want to know if it's a person, you now have to: 
1. call `getTypeId`
2. call `getTypeById`
3. call `isPersonType`

Do you think a shortcut for these three calls is not worth for this interface?

Reply:
Worth it! I mean to remove `isPersonType` since it doesn't handle when the type is an object. It's only used in Target Aggregate Service 's getSupervisedContacts anyway. 

Then keep this `isPerson(` as the only public function:
```
async isPerson(contact) {
  let type;
    if (typeof contact.type === 'object') {
      type = contact.type;
    } else {
      type = await this.get(this.getTypeId(contact));
    }
    return contactTypesUtils.isPersonType(type);
  }
```

Reply:
Or `isPersonType` can be private

Reply:
> getSupervisedContacts anyway

I checked out this code and we still need `isPersonType` there. 
I'm leaving it as is.

Reply:
Ok, I refactored it and added a `getPersonChildTypes` so we don't have type-related code in the target aggregate service

Reply:
Because in `getSupervisedContacts` the `type` is string, and `async isPerson(contact) {` doesn't handle the case when `type` is provided as string, right?

Reply:
yea. I changed it in a different way. it's better now.